app/stack/log-drains/new/route.js

```diff
@@ -2,11 +2,26 @@ import Ember from 'ember';
 import DS from 'ember-data';
 
 export default Ember.Route.extend({
+
   model: function(){
-    return this.store.createRecord('log-drain');
+    var stack = this.modelFor('stack');
+    return Ember.RSVP.all([
```

Comment:
I think using `Ember.RSVP.hash` is a little better as you can name the promises and then update `setupController` method to use names rather than indexes.  e.g. https://github.com/aptible/dashboard.aptible.com/blob/586767eef70c4b158241e9e9ff4145d6dc9b58d5/app/organization/roles/index/route.js#L7